inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/SortSdkSource.java

```diff
@@ -132,30 +136,78 @@ private void initReloadExecutor() {
+    private void reloadAll() {
 
         final List<SortTaskConfig> configs = SortClusterConfigHolder.getClusterConfig().getSortTasks();
         LOG.info("start to reload SortSdkSource");
+        this.startNewClients(configs);
+        this.stopExpiryClients(configs);
+        this.updateAllClientConfig();
+    }
 
-        // Start new clients
-        for (SortTaskConfig taskConfig : configs) {
-
-            // If exits, skip.
-            final String sortId = taskConfig.getName();
-            SortClient client = this.clients.get(sortId);
-            if (client != null) {
-                continue;
-            }
-
-            // Otherwise, new one client.
-            client = this.newClient(sortId);
-            if (client != null) {
-                this.clients.put(sortId, client);
-            }
-        }
+    /**
+     * Start a new client from SortTaskConfig.
+     * <p>
+     *     If the sortId is in configs, but not in active clients, start it.
+     * </p>
+     *
+     * @param configs Updated SortTaskConfig
+     */
+    private void startNewClients(final List<SortTaskConfig> configs) {
+        configs.stream()
+                .map(SortTaskConfig::getName)
+                .filter(sortId -> !clients.containsKey(sortId))
+                .forEach(sortId -> {
+                    final SortClient client = this.newClient(sortId);
+                    Optional.ofNullable(client)
+                            .ifPresent(c -> clients.put(sortId, c));
+                });
+    }
+
+    /**
+     * Stop an expiry client from SortTaskConfig.
+     * <p>
+     *     If the sortId is not in active clients, but not in configs, stop it.
+     * </p>
+     *
+     * @param configs Updated SortTaskConfig
+     */
+    private void stopExpiryClients(final List<SortTaskConfig> configs) {
+        Set<String> updatedSortIds = configs.stream()
+                .map(SortTaskConfig::getName)
+                .collect(Collectors.toSet());
+
+        clients.keySet().stream()
+                .filter(updatedSortIds::contains)
+                .forEach(sortId -> {
+                    final SortClient client = clients.get(sortId);
+                    try {
+                        client.close();
+                    } catch (Throwable th) {
+                        LOG.error("Got a throwable when close client {}, {}", sortId, th.getMessage());
+                    }
+                    clients.remove(sortId);
+                });
+    }
+
+    /**
+     * Update all client config.
+     */
+    private void updateAllClientConfig() {
+        clients.values().stream()
+                .map(SortClient::getConfig)
+                .forEach(this::updateClientConfig);
+    }
+
+    /**
+     * Update one client config.
+     *
+     * @param config The config to be updated.
+     */
+    private void updateClientConfig(SortClientConfig config) {
+        config.setManagerApiUrl(CommonPropertiesHolder.getSourceConfigManagerUrl());
```

Comment:
"CommonPropertiesHolder.getSourceConfigManagerUrl()" do not support to periodiclly update.

Reply:
it will support in #3243